### PR TITLE
Add Cypress

### LINKED ISSUE #1650: Add an E2E test for timeline interactions
We fixed a timeline regression in #1632. It'd be nice to test basic functionality

- clicking / hovering on the timeline
- graphics updating
- seeking to a location
- pausing on a line of code
- zoom in / zoom out

This is a lot, so we can keep this focused based on how hard each item is... I think we can simulate all the events though like this

```js
var event = new MouseEvent('mouseover', {
  'view': window,
  'bubbles': true,
  'cancelable': true
});

element.dispatchEvent(event);
```

### PR DESCRIPTION
**Pull request by jaril**
Fix #1650.

This doesn't work yet because it's missing the proper `.env` and `cypress.env.json` variables. That'll be a follow up.

References:
https://community.auth0.com/t/auth0-react-and-cypress/47855
https://stackoverflow.com/questions/59664721/how-to-test-single-page-application-with-cypress-and-auth0/59716793#59716793
https://github.com/auth0/auth0-spa-js/issues/210
https://stackoverflow.com/questions/50820732/in-cypress-set-a-token-in-localstorage-before-test
https://github.com/adamjmcgrath/cypress-spa-example/issues/2

### CODE REVIEW

File: cypress/integration/comments_spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+describe("login", () => {
```
Comment (jasonLaster) [inline cypress/integration/comments_spec.js:1]: could we move this into, `test/cypress`

File: package.json [hunk 2]
```diff
@@ -32,6 +33,7 @@
     "classnames": "^2.2.6",
     "codemirror": "^5.58.2",
     "css-loader": "^3.5.3",
+    "cypress-real-events": "^1.2.0",
```
Comment (jasonLaster) [inline package.json:36]: could this go in `devDependencies`?

File: src/ui/components/App.tsx [hunk 3]
```diff
@@ -47,6 +47,9 @@ function installViewportObserver({ updateNarrowMode }: Pick<AppProps, "updateNar
 
 function App({ theme, recordingId, modal, updateNarrowMode }: AppProps) {
   const auth = useAuth0();
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => setMounted(true), []);
```
Comment (jasonLaster) [inline src/ui/components/App.tsx:52]: what is this doing?

File: src/ui/components/Comments/CommentThread.js [hunk 4]
```diff
@@ -92,7 +93,14 @@ function Comment({ comment, isRoot, hoveredComment, pendingComment, setPendingCo
   }
 
   return (
-    <div className="comment-body-item">
+    <div
+      className={classnames("comment-body-item", { hover: isHovered })}
+      onMouseEnter={() => {
+        console.log("onmouseenter");
```
Comment (jasonLaster) [inline src/ui/components/Comments/CommentThread.js:99]: :salt:

File: src/ui/utils/tokenManager.tsx [hunk 5]
```diff
@@ -6,10 +6,20 @@ import { useHistory } from "react-router-dom";
...
+let clientId: string;
+const isTesting = new URL(window.location.href).searchParams.get("e2etest");
+
+if (isTesting) {
+  // Auth0 Application: Cypress E2E Testing
+  clientId = "uL3A8jxVqotF5Q6bmn5QTV46hU97MPQm";
+} else {
+  // Auth0 Application: Replay
+  clientId = "4FvFnJJW4XlnUyrXQF8zOLw6vNAH1MAo";
+}
```
Comment (jasonLaster) [inline src/ui/utils/tokenManager.tsx:21]: nit, but leets use a ternary and drop cypress... we'll need this without cypress as well

```
// Auth0 E2E tests require a different clientId
const clientId = isTesting ? <> : <>;
```

**Review by jasonLaster — COMMENTED** [review]
A couple of tiny comments

File: src/ui/components/App.tsx (hunk 3 above)
Comment (jaril) [reply src/ui/components/App.tsx:52]: Leftover code — just took it out.

File: cypress/integration/comments_spec.js (hunk 1 above)
Comment (jaril) [reply cypress/integration/comments_spec.js]: Done!

**Pull request merged**